libraries/AC_SafeRTL/AC_SafeRTL.cpp

```diff
@@ -152,33 +143,49 @@ bool SafeRTL_Path::pop_point(Vector3f& point)
     return false;
 }
 
-void SafeRTL_Path::reset_path(const Vector3f& start)
+/**
+*   Pass nullptr for the gcs argument if you don't want errors being reported to the user.
+*/
+void SafeRTL_Path::reset_path(bool position_ok, const Vector3f start, GCS* gcs)
```

Comment:
You do not need to pass gcs around.

gcs() gives you the GCS singleton object, pretty much anywhere in the code.

So `gcs()->send_text(MAV_SEVERITY_EVERYBODY_DIES, "Please press the green button");`

You may need to `#include <GCS_MAVLink/GCS.h>`

There are a whole bunch of checks in here which say `if (gcs)`; you don't need those either, just use the singleton.